src/db.c

```diff
@@ -449,7 +449,11 @@ static void dbSetValue(redisDb *db, robj *key, robj **valref, int overwrite, dic
         updateKeysizesHist(db, slot, kvNew->type, -1, newlen);
     }
 
-    if (server.lazyfree_lazy_server_del) {
+    /* In multi-threaded mode, the OBJ_ENCODING_RAW string object usually is
+     * allocated in the IO thread, so we defer the free to the IO thread. */
```

Comment:
move these comments into `if (server.io_threads_num > 1 && old->encoding == OBJ_ENCODING_RAW)`?

Reply:
even `lazyfree_lazy_server_del` was enabled, string obj was still freed in main thread.
Is it necessary to comment and explain it? Otherwise, this modification might make people think that if it is not handled by the io thread, it should also be handled in bio.

Reply:
for your fist comment, makes sense, i can do that.

for the second comment, i don't think this modification brings this confusion. this change only adds handling for an additional case, all other parts are unchanged. Maybe people need to say the comments of `freeObjAsync` in which we release the string type objects in the main thread.

btw, if a string object is not released by IO threads, so the object should be created by main thread, it might be more lightweight to release it in the main thread than in a BIO thread.

Reply:
maybe i don't understand your point, i added a comment, please have a look.